docs/design/crd-proposal-phase1.md

```diff
@@ -0,0 +1,220 @@
+* `mode` - enumerated string "user" or "active-directory" - Indicates that this is an AD joined instance or based on a local mapping of users
+* `users` - mapping - A new subsection pertaining to "user" options - required when mode = "user"
+  * `secret` - string - The name of a secret containing user and group definitions that will be used to create the local users and groups for the service
+* `localName` - string - Optional string used to indicate how this instance will
+  be referred to in an Active Directory
+* `realm` - string - The dns domain name of an Active Directory domain
+* `domains` - list-of-maps - A new subsection pertaining to detailed "active-directory" configuration (optional)
+  * `backend` - enumerated string - the name of a (supported) ID mapping backend
+  * `name` - string - the name of the domain being configured
+  * TBD - options to configure the backend, ID range, etc.
+
+
+## SmbShare
+
+Spec Options:
+* `shareName` - string - Optional string giving the share a full SMB compliant name. If unset, the name will be derived by the operator.
+* `storage` - mapping - A new subsection to configure the storage backing the share.
+    * `pvc` - mapping - A new subsection used to configure PVC backing storage.
+    * `name` - string - The name of a PVC, managed externally to the operator, for the share
+    * `spec` - mapping, embedded pvc spec - An embedded PVC spec that will be used to dynamically create a backing PVC for the share; sharing the lifecycle of the PVC with the share
+  * TBD - Any other more custom storage backends if needed
+* `securityDomain`  - string - The name of the SmbSecurityDomain CR associated with this share
+* `grouping` - string - Optional string. When one or more SmbShare CRs contain the same gropuing value it is an indication that they should be grouped together. If undefined it will be generated by the operator. The grouping is used by the operator as a hint to combine shares under one smbd service, but only as a hint as the ability to group under one smbd depends on other settings
+* `clustering` - mapping - A new subsection pertaining to clustering smbd instances
+  * `mode` - enumerated string "none" or "ctdb" - This option tells the operator if this share should be clustered or not
+  * `minSize` - int - Minimum number of smbd instances when clustering
+  * TBD - other clustering specific options
+* `customConfig` - mapping - A new subsection used to load "non-supported" settings
+   * `name` - Name of a configmap.
+   * TBD - we need to define this further
+* `readOnly` - boolean - When true this share will be read-only (default: false) [1]
+* `browseable` - boolean - When false this share will not be browsable (default: true)
+
+## Examples
+
+Two shares sharing the same "local" user mapping:
+```yaml
+---
+apiVersion: samba-operator.samba.org/v1alpha1
+kind: SmbSecurityDomain
+metadata:
+  name: "local-users"
+spec:
+  mode: "user"
+  users:
+    secret: "local-users-secret"
```

Comment:
Just curious: What is that secret used for?

Reply:
I envision it would contain a json configuration of local users and groups, the actual structure of said json TBD of course. :-)